exercises/protein-translation/protein_translation_test.go

```diff
@@ -0,0 +1,63 @@
+package protein
+
+import (
+	"testing"
+)
+
+const targetTestVersion = 1
+
+type testCase struct {
+	input    string
+	expected string
+}
+
+var codonTestCases = []testCase{
+	{"AUG", "Methionine"},
+	{"UUU", "Phenylalanine"},
+	{"UUC", "Phenylalanine"},
+	{"UUA", "Leucine"},
+	{"UUG", "Leucine"},
+	{"UCU", "Serine"},
+	{"UCC", "Serine"},
+	{"UCA", "Serine"},
+	{"UCG", "Serine"},
+	{"UAU", "Tyrosine"},
+	{"UAC", "Tyrosine"},
+	{"UGU", "Cysteine"},
+	{"UGC", "Cysteine"},
+	{"UGG", "Tryptophan"},
+	{"UAA", "STOP"},
+	{"UAG", "STOP"},
+	{"UGA", "STOP"},
+}
+
+var proteinTestCases = []testCase{
+	{"AUGUUUUCUUAAAUG", "Methionine Phenylalanine Serine"},
+	{"AUGUUUUGG", "Methionine Phenylalanine Tryptophan"},
+	{"AUGUUUUAA", "Methionine Phenylalanine"},
+	{"UGGUGUUAUUAAUGGUUU", "Tryptophan Cysteine Tyrosine"},
+}
+
+func TestCodon(t *testing.T) {
+	for _, test := range codonTestCases {
+		actual := proteinLookUp(test.input)
```

Comment:
Small question here - in most of our exercises we have the functions under test as exported, here it is unexported. Are we OK with this? Is there a specific argument we have for why things should be one way or the other? The only I could find was in the README "In general, adding tests for unexported names is considered an anti-pattern" (with an exception for the `testVersion`)
(But if we do change it to exported, it will be called as `protein.ProteinLookUp` which will be repetitive so if we export it it can just be called `Lookup` or something similar)